### DISCUSSION TITLE
How to stop bot from suggesting updates to SNAPSHOT versions or not released yet?

### DISCUSSION THREAD

**Discussion by jonathanbms** [renovatebot/renovate#14506] [Request Help]
I really like the bot, however it keeps suggesting updates that I don't want. How to fix this?

Example (bot suggesting an update to a snapshot version):
![image](https://user-images.githubusercontent.com/6158171/156657278-f598378a-ab5f-4545-ad63-211e9ab4a0b8.png)

Example (bot suggesting an update to an unfinished version):
![image](https://user-images.githubusercontent.com/6158171/156657402-db78459b-f195-4d28-86a2-cf254a8c75ea.png)

I would like the bot to suggest only official, stable releases of dependencies. Example:

Dependency A:
installed version: 1.4.0
suggested version: 1.4.1

Suggestions?

**Comment by rarkins** [comment]
You can switch to using semver versioning (which is essentially what you describe) however please note:
- Not all maven packages support this, so some might not work, and
- You already have e.g. 3.0.0-M5 in one example so your existing code does not comply with it

Note that unlike in semver, "1.0.0-fabcdef" does not mean "unstable". Renovate otherwise has a default behavior that it won't upgrade from stable to unstable, but in this case it's not meeting the criteria of unstable version from a maven spec.

**Reply by rarkins** [reply]
BTW please read https://octopus.com/blog/maven-versioning-explained

According to this:
- "M5" is not a stable version
- "SNAPSHOT" comes after "M5"

Therefore if that's true:
- You are already using a non-stable version
- Renovate is correct in attempting to upgrade you from milestone to snapshot

So to say you only want stable versions is contradicted by the fact that you're already using some unstable. No automated process can possibly read your mind and decide "this project has an existing unstable version but I'm going to guess they don't want any more", so if this is the case then you need to override it with config, by e.g. setting versioning=semver in a package rule.
